### PR TITLE
r_d17: Vector2d, work with operators

### PR DESCRIPTION
**Pull request by ShekhaninKyryl**
Please review this PR: 

```
Створити Vector2d(аналог математичного двовимірного вектора) з двома полями даних: float x, y зі значення по замовчуванню 0.0; 0.0; реалізувати 

operator=;
Vector2d operator+(const Vector2d& secondVector), Vector2d operator-(const Vector2d& secondVector) : відповідні компоненти двох векторів додаються/віднімаються, та повертається новий вектор; 
Vector2d operator+(const Vector2d& leftVector, const Vector2d& rightVector), Vector2d operator-(const Vector2d& leftVector, const Vector2d& rightVector) як дружні функції;
void operator*=(float scalar) : операція множення вектора на скаляр (https://ua.onlinemschool.com/math/library/vector/multiply3/)
float operator()() : повертає довжину вектора; (https://ua.onlinemschool.com/math/library/vector/)
float& operator[](std::size_t idx): для індексу 0 - повертає x, для значення 1 - повертає y;
Реалізувати метод для виводу вектора у консоль у форматі {x; y}
Приклад використання: Vector2d testVec{1.2, 5.6}; std::cout << testVec << std::endl; // Output: {1.2; 5.6}

Метод для вводу(інпуту) вектора у форматі x y (координати через пробіл) :
Приклад використання: std::cin >> testVec; 

Реалізувати підрахунок створених об’єктів(інстенсів) класу Vector2d (ті, що були створені та ще не видалені); Публічний константний метод класу для отримання кількості активних(живих) інстенсів.
```

### CODE REVIEW

File: r_d17/Vector2d.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	Vector2d operator=(const Vector2d& vector);
+	Vector2d operator+(const Vector2d& secondVector);
+	Vector2d operator-(const Vector2d& secondVector);
+
+	friend Vector2d operator+(const Vector2d& leftVector, const Vector2d& rightVector);
+	friend Vector2d operator-(const Vector2d& leftVector, const Vector2d& rightVector);
+
+	void operator*=(float scalar);
+	float operator()();
```
Comment (bohdanpc) [inline r_d17/Vector2d.h:24]: can be const

File: r_d17/Vector2d.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+float Vector2d::operator()()
+{
+	return static_cast<float>(std::sqrt(x * x + y * y));
+}
+
+float& Vector2d::operator[](std::size_t idx)
+{
+	if (idx == 0) return x;
+	if (idx == 1) return y;
```
Comment (bohdanpc) [inline r_d17/Vector2d.cpp:69]: no default return statement? I bet the compiler has warned you about that

File: r_d17/Vector2d.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+std::ostream& operator<<(std::ostream& out, Vector2d& vector)
+{
+	out << std::fixed << std::setprecision(1) << "Output: {" << vector.x << "; " << vector.y << "}";
```
Comment (bohdanpc) [inline r_d17/Vector2d.cpp:74]: you shouldn't apply manipulator's here. 
If someone would do that to make the output more precise: `std::cout << std::setprecision(20) << MyPreciseVec` he won't be able to as you override it unconditionally inside.

File: r_d17/r_d17.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+	/*
+	* 3
+	*/
+	std::cout << "Counter: " << Vector2d::getCounter() << '\n';
+
+	/*
+	* 2
+	*/
+	vector1.~Vector2d();
```
Comment (bohdanpc) [inline r_d17/r_d17.cpp:57]: no explicit destructor call is allowed in general

File: r_d17/Vector2d.cpp (hunk 2 above)
Comment (ShekhaninKyryl) [reply r_d17/Vector2d.cpp:69]: nope, no warning from compiler)

**Pull request merged**